### PR TITLE
6 Re-establish appcenter builds for meditrak

### PR DESCRIPTION
**Pull request by edmofro**
### Issue #:
https://github.com/beyondessential/tupaia-backlog/issues/6

Code changes to go along with manual setup on appcenter.ms

### CODE REVIEW

File: packages/meditrak-app/app/widgets/BlueButton.js [hunk 1]
```diff
@@ -4,7 +4,7 @@
  */
 import React from 'react';
 import { StyleSheet, View } from 'react-native';
-import { Button } from '.';
```
Comment (edmofro) [inline packages/meditrak-app/app/widgets/BlueButton.js:7]: This was a stab in the dark when the build was breaking, but it worked. Seems something about the appcenter build doesn't like to import from the index.js of the same directory

File: packages/meditrak-app/android/appcenter-gradle.properties [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# mustn't contain secrets like the keystore password, but can contain gradle config switches
+
+# AndroidX package structure to make it clearer which packages are bundled with the
+# Android operating system, and which are packaged with your app's APK
+# https://developer.android.com/topic/libraries/support-library/androidx-rn
+android.useAndroidX=true
+# Automatically convert third-party libraries to use AndroidX
+android.enableJetifier=true
+
+org.gradle.jvmargs=-XX\:MaxHeapSize\=1024m -Xmx1024m
```
Comment (edmofro) [inline packages/meditrak-app/android/appcenter-gradle.properties:11]: This file will get renamed to `gradle.properties` in appcenter (see the pre-build script)

File: packages/meditrak-app/ios/TupaiaMediTrak.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -779,7 +779,7 @@
 				PRODUCT_BUNDLE_IDENTIFIER = com.tupaia.meditrak;
 				PRODUCT_NAME = TupaiaMediTrak;
 				PROVISIONING_PROFILE = "75127d56-33b6-49a5-9318-47d4bb76efdf";
-				PROVISIONING_PROFILE_SPECIFIER = "Tupaia Distribution Profile 2020 - 2021";
+				PROVISIONING_PROFILE_SPECIFIER = "Tupaia Distribution Profile 2021 - 2022";
```
Comment (edmofro) [inline packages/meditrak-app/ios/TupaiaMediTrak.xcodeproj/project.pbxproj:782]: Got ahead of our upcoming provisioning profile expiry

File: packages/meditrak-app/appcenter-post-clone.sh [hunk 4]
```diff
@@ -1,15 +1,28 @@
...
+cd ../..
+
+# install root dependencies
+SKIP_BUILD_INTERNAL_DEPENDENCIES=true yarn install
+
+# move to meditrak folder
+cd packages/meditrak-app
+
+# build internal dependencies of meditrak
+../../scripts/bash/buildInternalDependencies.sh --packageJsonPath ./package.json
```
Comment (edmofro) [inline packages/meditrak-app/appcenter-post-clone.sh:24]: Building ui-components etc. was totally unnecessary and taking ages so I updated buildInternalDependencies to only build deps of a given package.json (separate PR, see https://github.com/beyondessential/tupaia/pull/2474)

**Review by rohan-bes — APPROVED** [review]
Looks real good to me! Very nice of you to post the solution you made to the origin ticket 👍

File: scripts/bash/backendStartDev.sh [hunk 5]
```diff
@@ -26,10 +26,14 @@ while [ "$2" != "" ]; do
         type_script=true
         shift
         ;;
-    -s | --skip-internal)
+    -i | --include-internal)
         skip_internal=true
```
Comment (rohan-bes) [inline scripts/bash/backendStartDev.sh:30]: I think you need to rename this to `inclue_internal`?

File: scripts/bash/backendStartDev.sh [hunk 6]
```diff
@@ -56,4 +57,7 @@ else
     # many restarts that otherwise happen during the initial build of internal dependencies
     start_server="${start_server} --delay 1 ${watch_flags}"
     yarn concurrently "${DIR}/buildInternalDependencies.sh --watch --packagePath ." "eval ${start_server}"
+else
+    echo "Starting server without internal dependency build and watch. To include internal dependencies, add the -i flag - it's much faster than it used to be!"
```
Comment (rohan-bes) [inline scripts/bash/backendStartDev.sh:61]: Hehehe I love the positivity of this info message 😻

**Review by rohan-bes — CHANGES_REQUESTED** [review]

File: scripts/bash/backendStartDev.sh (hunk 5 above)
Comment (edmofro) [reply scripts/bash/backendStartDev.sh]: Eeek, thanks for the review!! 👀

File: scripts/bash/backendStartDev.sh [hunk 7]
```diff
@@ -44,16 +48,16 @@ fi
 
 echo "Starting server"
 
-if [[ ${skip_internal} == true ]]; then
-    echo "Skipping internal dependency build and watch"
-    eval ${start_server}
-else
+if [[ ${inclue_internal} == true ]]; then
```
Comment (rohan-bes) [inline scripts/bash/backendStartDev.sh:51]:
```suggestion
if [[ ${include_internal} == true ]]; then
```

Comment (edmofro) [reply scripts/bash/backendStartDev.sh]: Thanks again! I've been through the script more carefully now, and actually tested the positive case (had only tested the negative ones to make sure my friendly messages were printing before 🤦 )

**Review by rohan-bes — APPROVED** [review]
Looks good to me!
